src/network.rs

```diff
@@ -29,6 +31,47 @@ static DEFAULT_TIMEOUT_SECS: u64 = 60;
 
 const MAX_CONCURRENT_CONNECTIONS: usize = 40;
 
+/// The network payload encoding.
+///
+/// This is only used in `download` (not `download_and_persist`) because (for now) it's only needed
+/// in downloading imports (not packages, which is what `download_and_persist` is used for) to
+/// workaround known server shortcomings. It could be added to `download_and_persist`, but due to
+/// the use of `tokio::io::File` it gets messy and either this or that would need a refactor.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PayloadEncoding {
+    Plaintext,
+    Base64,
+}
+
+impl From<&Response> for PayloadEncoding {
+    fn from(response: &Response) -> Self {
+        // gitiles always encodes content in base64
+        if response.headers().contains_key("x-gitiles-object-type") {
+            Self::Base64
+        } else {
+            Self::Plaintext
+        }
+    }
+}
```

Comment:
nit: as this is never really used with `into`, I'd probably just write this as a normal constructor like `fn for_response(response: &Response) -> Self {` rather than a `from` impl, as that lets you give more context.

Reply:
Sure. I usually add `From` impls if a type can be constructed from a single other type, but I don't mind adding an explicit method.